d2layouts/d2elklayout/layout.go

```diff
@@ -374,6 +374,7 @@ func Layout(ctx context.Context, g *d2graph.Graph, opts *ConfigurableOpts) (err
 		if hasTop || hasBottom {
 			padding := parsePadding(elkNodes[obj].LayoutOptions.Padding)
 			if hasTop {
+				// TODO I think this fails to account for a potential inner label of container
```

Comment:
@gavin-ts there is currently a collision between the icon and label in ELK. I believe it's unrelated to my change and that this code has a bug as described. Can you confirm if so or if I'm just missing something.

<img width="1401" alt="Screen Shot 2023-12-11 at 4 16 05 PM" src="https://github.com/terrastruct/d2/assets/3120367/b9032857-c15b-41dd-b8e3-92d2fbbc6a10">

Reply:
yeah it looks like elk hasn't been updated for this. previously all icon positions would be placed in an inside position by default so this would only be possible with custom set positions. 

not sure why but elk only seems to be adjusting horizontally for icons but not vertically

https://github.com/terrastruct/d2/blob/56a549d7c197f33e4c4fb2f88af6db5dab1d7869/d2layouts/d2elklayout/layout.go#L1125-L1131

also noticed elk layout is still doing some old height manipulations but we already look for intersections with outside labels so these are no longer needed

https://github.com/terrastruct/d2/blob/56a549d7c197f33e4c4fb2f88af6db5dab1d7869/d2layouts/d2elklayout/layout.go#L1184-L1187

https://github.com/terrastruct/d2/blob/56a549d7c197f33e4c4fb2f88af6db5dab1d7869/d2layouts/d2elklayout/layout.go#L1112-L1115



https://github.com/terrastruct/d2/blob/56a549d7c197f33e4c4fb2f88af6db5dab1d7869/d2graph/layout.go#L439-L445

Reply:
in the elk `icon_positions` test it appears this is unaccounted for prior to your PR, it also may not be accounted for in dagre if the label were inside

<img width="1271" alt="Screenshot 2023-12-11 at 5 38 05 PM" src="https://github.com/terrastruct/d2/assets/85081687/126fedb4-d854-4fa8-91fb-7634442dde86">

Reply:
okay, i'll get in followup then

Reply:
@gavin-ts 
are you sure about the "these are no longer needed"? Many tests change when I remove those:

https://github.com/terrastruct/d2/pull/1775/commits/8ce19c551693a40b884b34a9ea94d95a6945e0c7

Reply:
in that commit, you also removed the code that made the other 2 unnecessary

Reply:
i reverted, there's still changes: https://github.com/terrastruct/d2/pull/1775/commits/705225ee7948d1db9c88f9feb850a1e89371afc6

Reply:
https://github.com/terrastruct/d2/commit/8ce19c551693a40b884b34a9ea94d95a6945e0c7#r134821504

looks like we do want the || icon part for this 
<img width="757" alt="Screenshot 2023-12-12 at 10 46 19 AM" src="https://github.com/terrastruct/d2/assets/85081687/febff510-8656-4881-b604-16c6a79d86f8">

Reply:
I think we may actually want this adjustment for all outside labels/icons unless we adjust the spacing in another way
https://github.com/terrastruct/d2/commit/8ce19c551693a40b884b34a9ea94d95a6945e0c7#r134822027

to avoid this
<img width="502" alt="Screenshot 2023-12-12 at 10 49 31 AM" src="https://github.com/terrastruct/d2/assets/85081687/988411c9-3c46-4dc5-9c07-ac2590149369">

Reply:
is this captured in https://github.com/terrastruct/d2/pull/1776 or can you make another issue?

Reply:
yes this is captured in #1776, all margins are now included in the width/height passed to elk, so elk ends up routing the edge distance to shape + margin, instead of routing to shape then cutting at margin.